packages/kolibri-common/components/SyncSchedule/ManageSyncSchedule.vue

```diff
@@ -0,0 +1,365 @@
+            <div v-else>
+              <div class="loader-size">
+                <KCircularLoader
+                  :delay="false"
+                  class="loader"
+                  :size="18"
+                />
+              </div>
+            </div>
+          </KGridItem>
+
+          <KGridItem
+            :layout="{ alignment: 'right' }"
+            :layout8="{ span: 4 }"
+            :layout12="{ span: 6 }"
+          >
+
+            <KButton
+              appearance="basic-link"
+              :text="$tr('forgetText')"
+            />
+          </KGridItem>
+        </KGrid>
+      </KModal>
+      <AddAddressFormVue v-if="newaddressclick" />
+    </KPageContainer>
+  </ImmersivePage>
+
+</template>
+
+
+<script>
+
+  import ImmersivePage from 'kolibri.coreVue.components.ImmersivePage';
+  import CoreTable from 'kolibri.coreVue.components.CoreTable';
+  import { TaskResource, FacilityResource, NetworkLocationResource } from 'kolibri.resources';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import { PageNames } from '../../../../kolibri/plugins/facility/assets/src/constants';
+  import AddAddressFormVue from '../../../../kolibri/core/assets/src/views/sync/SelectAddressModalGroup/AddAddressForm.vue';
+
+  export default {
+    name: 'ManageSyncSchedule',
+    components: {
+      ImmersivePage,
+      CoreTable,
+      AddAddressFormVue,
+    },
+    extends: ImmersivePage,
+    mixins: [commonCoreStrings],
+    data() {
+      return {
+        deviceModal: false,
+        facility: null,
+        data: null,
+        radioBtnValue: ' ',
+        newaddressclick: false,
+        deviceIds: [],
+        myDevices: [],
+      };
+    },
+    computed: {
+      goBack() {
+        return { name: PageNames.DATA_EXPORT_PAGE };
+      },
+    },
+    beforeMount() {
+      this.pollFacilityTasks();
+      this.fetchFacility();
+      this.fetchAddressesForLOD();
+    },
+
+    methods: {
+      fetchFacility() {
+        FacilityResource.fetchModel({ id: this.$store.getters.activeFacilityId, force: true }).then(
+          facility => {
+            this.facility = { ...facility };
+          }
+        );
+      },
+      fetchAddressesForLOD(LocationResource = NetworkLocationResource) {
+        return LocationResource.fetchCollection({ force: true }).then(locations => {
+          this.data = locations;
+        });
+      },
+      pollFacilityTasks() {
+        TaskResource.list({ queue: 'facility_task' }).then(tasks => {
+          this.myDevices = tasks;
+
+          if (this.isPolling) {
+            setTimeout(() => {
+              return this.pollFacilityTasks();
+            }, 2000);
+          }
+        });
+      },
+      closeModal() {
+        this.deviceModal = false;
+      },
+      submitModal(id) {
+        this.deviceModal = false;
+        if (id !== ' ') {
+          this.deviceIds.push(id);
+
+          this.$router.push({ path: '/editdevice/?id=' + id });
```

Comment:
To make this work with the `deviceId` url parameter, you would need to update this (can pass parameters as part of [programmatic navigation](https://v3.router.vuejs.org/guide/essentials/navigation.html)):
```
 this.$router.push({ name: PageNames.EDIT_SYNC_SCHEDULE, params: { deviceId: id } });
```
Once you've defined the `EDIT_SYNC_SCHEDULE` page name and used that for the route.

You would also need to update the EditSyncSchedule component to read the deviceId from params.deviceId rather than from the query.